common/src/main/java/org/apache/celeborn/common/network/protocol/TransportMessage.java

```diff
@@ -97,13 +99,10 @@ public <T extends GeneratedMessageV3> T getParsedPayload() throws InvalidProtoco
   }
 
   public ByteBuffer toByteBuffer() {
-    int totalBufferSize = payload.length + 4 + 4;
-    ByteBuffer buffer = ByteBuffer.allocate(totalBufferSize);
-    buffer.putInt(messageTypeValue);
-    buffer.putInt(payload.length);
-    buffer.put(payload);
-    buffer.flip();
-    return buffer;
+    ByteBuf headerBuf = Unpooled.buffer().writeInt(messageTypeValue).writeInt(payload.length);
```

Comment:
`Unpooled.buffer()` allocates 256 bytes as default `static final int DEFAULT_INITIAL_CAPACITY = 256;`, which is much larger than the needed 8 bytes. Additionally, `toByteBuffer` is used for few messages whose payload is not big, IMO it's OK to leave what it is.